MoppLib/MoppLib/PublicInterface/MoppLibCryptoActions.m

```diff
@@ -39,7 +48,78 @@ + (MoppLibCryptoActions *)sharedInstance {
     });
     return sharedInstance;
 }
+
+- (void)parseCdocInfo:(NSString *)fullPath success:(CdocContainerBlock)success failure:(FailureBlock)failure {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        NSError *error;
+        CdocInfo *response;
+         @try {
+            CdocParser *cdocParser = [CdocParser new];
+            response = [cdocParser parseCdocInfo:fullPath];
+            for (Addressee* addressee in response.addressees) {
+                MoppLibCerificatetData *certData = [MoppLibCerificatetData new];
+                NSData *certificate = addressee.cert;
+                NSString* certificateWithUTF8 = [NSString stringWithUTF8String:[certificate bytes]];
+                //Sometimes there may be a redundant line change
+                NSString *formattedCertificate = [certificateWithUTF8 stringByReplacingOccurrencesOfString:@"\n\n" withString:@"\n"];
+                NSData* decodedCertificate = [formattedCertificate dataUsingEncoding:NSUTF8StringEncoding];
+
+                [MoppLibCertificate certData:certData updateWithDataPem:[decodedCertificate bytes] length:decodedCertificate.length];
+                addressee.type = [self formatTypeToString :certData.organization];
+                addressee.validTo = certData.expiryDate;
+            }
+         }
+        @catch (...) {
+            error = [MoppLibError generalError];
+        }
+        dispatch_async(dispatch_get_main_queue(), ^{
+            error == nil ? success(response) : failure(error);
+        });
+    });
+}
+
+- (NSString*)formatTypeToString:(MoppLibCertificateOrganization)formatType {
+    NSString *result = nil;
+    switch(formatType) {
+        case DigiID:
+            result = @"DIGI-ID";
+            break;
+        case IDCard:
+            result = @"ID-CARD";
+            break;
+        default:
+            result = @"E-SEAL";
+            break;
+    }
     
+    return result;
+}
+
+- (void)decryptData:(NSString *)fullPath withPin1:(NSString*)pin1 success:(DecryptedDataBlock)success failure:(FailureBlock)failure {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        NSError *error;
+        NSMutableDictionary *response;
+        @try {
+            Decrypt *decrypter = [Decrypt new];
+            SmartToken *smartToken = [SmartToken new];
+            response = [decrypter decryptFile:fullPath withPin:pin1 withToken:smartToken];
+        }
+        @catch (NSException *exception) {
+            if([[exception name] isEqualToString:@"Wrong PIN"]) {
```

Comment:
Equality check against human-readable text. Can't this be ID, like `@"wrong_pin"`?